ibis/expr/tests/test_table.py

```diff
@@ -695,6 +698,89 @@ def test_asof_join_with_by():
     assert by.left.op().name == by.right.op().name == 'key'
 
 
+@pytest.mark.parametrize(
+    ('ibis_interval', 'timedelta_interval'),
+    [
+        [ibis.interval(days=2), pd.Timedelta('2 days')],
+        [ibis.interval(days=2), datetime.timedelta(days=2)],
+        [ibis.interval(hours=5), pd.Timedelta('5 hours')],
+        [ibis.interval(minutes=7), pd.Timedelta('7 minutes')],
+        [ibis.interval(seconds=9), pd.Timedelta('7 seconds')],
+        [ibis.interval(seconds=9), datetime.timedelta(seconds=9)],
+        [ibis.interval(milliseconds=11), pd.Timedelta('11 milliseconds')],
+        [ibis.interval(microseconds=15), pd.Timedelta('15 microseconds')],
+        [ibis.interval(microseconds=15), datetime.timedelta(microseconds=15)],
+        [ibis.interval(nanoseconds=17), pd.Timedelta('17 nanoseconds')],
+        param(
+            ibis.interval(hours=5),
+            datetime.timedelta(hours=5),
+            id='dateime hours',
+            marks=pytest.mark.xfail(
+                reason='Hour conversion from datetime.timedelta to ibis '
+                'interval not supported'
+            ),
+        ),
+        param(
+            ibis.interval(minutes=7),
+            datetime.timedelta(minutes=7),
+            id='dateime minutes',
+            marks=pytest.mark.xfail(
+                reason='Minute conversion from datetime.timedelta to ibis '
+                'interval not supported'
+            ),
+        ),
+        param(
+            ibis.interval(milliseconds=11),
+            datetime.timedelta(milliseconds=11),
+            id='dateime milliseconds',
+            marks=pytest.mark.xfail(
+                reason='Millisecond conversion from datetime.timedelta to '
+                'ibis interval not supported'
+            ),
+        ),
+        param(
+            ibis.interval(weeks=3),
+            pd.Timedelta('3', unit='W'),
+            id='weeks',
+            marks=pytest.mark.xfail(
+                reason='Week conversion from Timedelta to ibis interval '
+                'not supported'
+            ),
+        ),
+        param(
+            ibis.interval(years=3),
+            pd.Timedelta('3', unit='Y'),
+            id='years',
+            marks=pytest.mark.xfail(
+                reason='Year conversion from Timedelta to ibis interval '
+                'not supported'
+            ),
+        ),
+    ],
+)
+def test_asof_join_with_tolerance(ibis_interval, timedelta_interval):
```

Comment:
I wonder if we can break these tests out into separate parts. testing the interval translation (probably moved to test_datatypes.py) and testing the asof_join with tolerance (here).